src/blocks/rofication.rs

```diff
@@ -0,0 +1,195 @@
+use std::str::FromStr;
+use std::time::Duration;
+
+use crossbeam_channel::Sender;
+use serde_derive::Deserialize;
+
+use std::io::Read;
+use std::io::Write;
+use std::os::unix::net::UnixStream;
+use std::path::Path;
+
+use crate::blocks::{Block, ConfigBlock, Update};
+use crate::config::SharedConfig;
+use crate::de::deserialize_duration;
+use crate::errors::*;
+use crate::protocol::i3bar_event::I3BarEvent;
+use crate::protocol::i3bar_event::MouseButton;
+use crate::scheduler::Task;
+use crate::subprocess::spawn_child_async;
+use crate::widgets::text::TextWidget;
+use crate::widgets::I3BarWidget;
+use crate::widgets::State;
+
+#[derive(Debug)]
+struct RotificationStatus {
+    num: u64,
+    crit: u64,
+}
+
+pub struct Rofication {
+    id: usize,
+    text: TextWidget,
+    update_interval: Duration,
+
+    //useful, but optional
+    #[allow(dead_code)]
+    shared_config: SharedConfig,
+    #[allow(dead_code)]
+    tx_update_request: Sender<Task>,
+    // UNIX socket to read from
+    pub socket_path: String,
```

Comment:
Can you add a `format` string too? It seems inevitable so we may as well do it from the start.

Reply:
> Can you add a format string too? It seems inevitable so we may as well do it from the start.

Done.